### ISSUE TITLE
Unable to sent mails (Gmail) TLS auth not started + ENV parameters ignored

### ISSUE THREAD

**Issue by gorjan-mishevski** [saleor/saleor#7420]
### What I'm trying to achieve
Sent mails from Gsuite smtp server.

EMAIL_URL
smtp://gorjan@|sensitive-data|:|sensitive-data|@smtp.gmail.com:587/?ssl=True
Added tls=True but it also fails

Double checked the docker env variables all of them are set (EMAIL_URL, DEFAULT_FROM_EMAIL....)

NOTE: Tested with online service smtp configuration is correct and sends mail, only Saleor fails to sent the email

### Steps to reproduce the problem
1. Create profile.
2. Set correct env variables.
3. Enable user emails plugin
4. Leave user email plugins empty.
5. Forget password flow.
6. Check logs, user email plugin is not reading the env variable

```
worker_1      | [2021-06-01 16:05:46,259: ERROR/ForkPoolWorker-1] Task saleor.plugins.user_email.tasks.send_password_reset_email_task[e403b132-5be3-4ace-8b54-9ae156dda655] raised unexpected: SMTPSenderRefused(530, b'5.7.0 Must issue a STARTTLS command first. b14sm108650edz.21 - gsmtp', 'support@|sensitve-data|')
worker_1      | Traceback (most recent call last):
worker_1      |   File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 405, in trace_task
worker_1      |     R = retval = fun(*args, **kwargs)
worker_1      |   File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 697, in __protected_call__
worker_1      |     return self.run(*args, **kwargs)
worker_1      |   File "/app/saleor/plugins/user_email/tasks.py", line 65, in send_password_reset_email_task
worker_1      |     send_email(
worker_1      |   File "/app/saleor/plugins/email_common.py", line 210, in send_email
worker_1      |     send_mail(
worker_1      |   File "/usr/local/lib/python3.9/site-packages/django/core/mail/__init__.py", line 61, in send_mail
worker_1      |     return mail.send()
worker_1      |   File "/usr/local/lib/python3.9/site-packages/django/core/mail/message.py", line 284, in send
worker_1      |     return self.get_connection(fail_silently).send_messages([self])
worker_1      |   File "/usr/local/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 109, in send_messages
worker_1      |     sent = self._send(message)
worker_1      |   File "/usr/local/lib/python3.9/site-packages/django/core/mail/backends/smtp.py", line 125, in _send
worker_1      |     self.connection.sendmail(from_email, recipients, message.as_bytes(linesep='\r\n'))
worker_1      |   File "/usr/local/lib/python3.9/smtplib.py", line 882, in sendmail
worker_1      |     raise SMTPSenderRefused(code, resp, from_addr)
worker_1      | smtplib.SMTPSenderRefused: (530, b'5.7.0 Must issue a STARTTLS command first. b14sm108650edz.21 - gsmtp', 'support@|sensitive-data|')
```

Snippet from the logs showing the default email provided is not respected.
It should not be "support@"
```
worker_1      | [2021-06-01 16:53:23,211: WARNING/ForkPoolWorker-4] 'from_email'
worker_1      | [2021-06-01 16:53:23,211: WARNING/ForkPoolWorker-4] :
worker_1      | [2021-06-01 16:53:23,211: WARNING/ForkPoolWorker-4] 'support@|sensitive-data|'
```
 I've found this from the logs also. It seems Saleor is ignoring the EMAIL_URL passed property.
```
EmailConfig(host='', port='', username='', password=None, sender_name='', sender_address='support@|sensitive-data|', use_tls=False, use_ssl=False)
```
Checking the account_customerevent table has the record with parameters set to {}. Which I think is not correct.

I have the following flag added EMAIL_HOST=smtp.gmail.com
which is loaded from Docker as env variable, but Saleor ignores it.
The dashboard has empty values, which means the flag should take precedence.

It seems this issue stems from the fact that the plugin settings even though they are empty they override the env variables. Because they were empty I had an empty configuration.

### What I expected to happen
To sent a mail after password reset.
And also sent it with the email from DEFAULT_FROM_EMAIL parameter

### Screenshots and logs

**System information**

Saleor version:
- [ ] dev (current master)
- [x ] 3.0
- [ ] 2.11
- [ ] 2.10

Operating system:
- [ ] Windows
- [ x] Linux
- [x ] MacOS
- [ ] Other

**Comment by tiendq** [comment]
Hi @gorjan-mishevski,

I have a same problem, can't send email with Gmail, could you tell me how to do steps from 3-6?

Thanks,

**Comment by acarlton** [comment]
We are having the same issue, currently using 3.0.0-b.3 release. Hosting on AWS, using SES for SMTP. I can confirm that emails send correctly through both the Fargate container operating system and the Django console.

I believe there is an issue with plugin configuration not correctly picking up the default settings from the environment variables. I found that I had to manually enable the user emails plugin for each relevant channel. The tooltips say that the default settings should be picked up from the environment variables. I have logged into the container and verified the settings values are correct using the Django console.

For a long time, I though I had misconfigured the celery task broker, but when I finally removed the CELERY_BROKER_URL from my configuration, I found that the API service tried to send the email and got an authorization error.

From there, I tried to manually input my SMTP settings into the plugin form and I was not able to due to strange validation errors from the GraphQL call. I believe there is an issue with the plugin configuration which isn't accepting the correct settings.

Please let me know if you have suggestions or need more information.

**Comment by korycins** [comment]
All configuration for UserEmails was moved to email plugin view. 
Only AdminEmail plugin uses`EMAIL_URL` (and system envs) when AdminEmail plugin configuration is empty. 
The one issue which I see is that you have an active UserEmail plugin with an empty configuration.

**Comment by korycins** [comment]
IMHO Introducing a new env which could contain config for a UserEmailPlugin  - like `USER_EMAIL_URL` would be a good idea.

**Comment by acarlton** [comment]
@korycins Yes, I can confirm that I was able to fix the user emails by manually entering the configuration values into the plugin form for each Channel.

Your suggestion to add a `USER_EMAIL_URL` variable sounds like a good idea. Also, the "help text" below each form element on the UserEmailPlugin form still indicates that it should default to the system environment values of `EMAIL_HOST`, `EMAIL_PORT` and `EMAIL_HOST_USER`. This should be changed to avoid confusion.

**Comment by korycins** [comment]
@acarlton  Thanks for letting me know. I didn't know about incorrect help text on UserEmailPlugin 👍

**Comment by simon-online** [comment]
I just spent time troubleshooting this issue which was totally unnecessary as this issue could have been resolved by now.

From my point of view the decision to not use EMAIL_URL server config as a fallback default for the UserEmailPlugin doesn't make any sense at all.

Furthermore it's nearly been 2 years and there is still no clear documentation anywhere about this and the "help text" below each form element on the UserEmailPlugin form still hasn't been updated.

There should be documentation in the following section to explain that EMAIL_URL is not actually used to send any user emails and this should also be configured in the UserEmailPlugin so user emails are successfully sent:
https://docs.saleor.io/docs/3.x/setup/configuration#email_url
